### ISSUE TITLE
Support TypedDict update() with keyword arguments

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6019]
Things like `d.update(x=1, y=2)` should be accepted if `d` is a TypedDict and the keys and values are compatible.

**Comment by emmatyping** [comment]
I think this would need to be added as a specification of the standard. I do not see this as allowed anywhere, and currently the update method is annotated as accepting only a `TypedDict` in typeshed.

Perhaps this should be moved to the typing repo?

**Comment by JelleZijlstra** [comment]
I don't think this would require a change to the spec. The spec is mostly silent on the `update()` method (there is some discussion from the ReadOnly PEP at https://typing.readthedocs.io/en/latest/spec/typeddict.html#update-method), but it wouldn't be against the spec to allow `update()` calls that preserve the type. Pyright also accepts it: https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjAFC1UA2AhgM6tQAaAFEQmZRoBKAFy0oEqMxGYUdWqWLAowbgA8ZnUeMlqAdAFcEpZjGLdmAXgCMQ2kA

> Perhaps this should be moved to the typing repo (or I guess CPython)?

The CPython issue tracker should only be used for tracking changes to the typing.py runtime. Spec changes should go to the python/typing issue tracker.
